### ISSUE TITLE
Enable filtering in galactic coordinates

### ISSUE THREAD

**Issue by nevencaplar** [astronomy-commons/lsdb#196]
Enable the user to specify the filtering regions (e.g., [box](https://github.com/astronomy-commons/lsdb/blob/25cf48b318e1a09983d26364f11b6b15f3a1ac47/src/lsdb/catalog/catalog.py#L206)), in galactic coordinates.

Possibly API might be (do they use ra/dec for galactic?):

`Catalog.box(ra=(340, 342), dec=(340, 342)), coordinates = 'galactic')`

The initial suggestion is to have the galactic coordinates box/region transformed into polygon in Equatorial Coordinate System that is sampled with a ``sufficent'' density. From there use existing functionality. 

In case that this turns out to be problematic, report back to consider alternative approaches (via e.g., astropy regions).

**Before submitting**
Please check the following:

- [X] I have described the purpose of the suggested change, specifying what I need the enhancement to accomplish, i.e. what problem it solves.
- [X] I have included any relevant links, screenshots, environment information, and data relevant to implementing the requested feature, as well as pseudocode for how I want to access the new functionality.
- [X] If I have ideas for how the new feature could be implemented, I have provided explanations and/or pseudocode and/or task lists for the steps.

**Comment by hombit** [comment]
Do you have a mathematic idea on how to build a great-circle polygon to contain a Galactic-coordinate box?

**Comment by hombit** [comment]
@nevencaplar one of the ways to do galactic latitude filtering `b` is cone-search centered in a galactic pole with a radius of `90 - |b|`

**Comment by nevencaplar** [comment]
@hombit Do you think this is ``simple enough'' to do it relatively quickly and ``easily''? @mjuric expressed a concern about implementing all different coordinates separately, opting rather for a single solution that could take in many more coordinate systems. Having said, galactic is probably a second most important system, and I am worried about that it might be quite a bit time before we are able to incorporate a general solution.

**Comment by hombit** [comment]
@nevencaplar Yes, I think it is "simple enough" even for an arbitrary coordinate system input (while all coordinates of the box corners are in the same system). All we need is to support two cases: "ICRS" and "other", while we could relatively easily implement "other" system if we have a way to convert it to ICRS
